packages/firestore/src/core/view.ts

```diff
@@ -301,9 +304,15 @@ export class View {
 
     this.applyTargetChange(targetChange);
     const limboChanges = updateLimboDocuments
-      ? this.updateLimboDocuments()
+      ? this.updateLimboDocuments(waitForRequeryResult ?? false)
       : [];
-    const synced = this.limboDocuments.size === 0 && this.current;
+
+    // We are at synced state if there is no limbo docs are waiting to be resolved, view is current
+    // with the backend, and the query is not pending for full re-query result due to existence
```

Comment:
nit: Consider changing "pending for full re-query" to "pending reset" because, IMO, it's clearer to use that more specific language.